### PR TITLE
feat: add cron parser tool

### PR DESCRIPTION
**Pull request by DotNetInAction**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] UI change (please include screenshot!)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Internationalization and localization
- [ ] Other (please describe):

## What is the current behavior?

Currently there is no tool to parse, validate and create Cron expressions in DevToys

Issue Number: #292

## What is the new behavior?

-New tool called Cron Parser has been added under the Generators section
-New tool allows user to create and validate Cron expression to see upcoming scheduled dates
-New tool supports 5 and 6 segment Cron (Without and With seconds), has ouput line limiter and Output date time formater with validation
-New tool runs on Local timezone to show user the correct output 
-It supports daylight saving's
-It is based on open-source library called Cronos

image

## Other information

Future modifications will include Cron Maker from the UI, cleaner explanation of Cron elements, and segments descriptions

## Quality check

Before creating this PR, have you:

- [x] Followed the code style guideline as described in [CONTRIBUTING.md](https://github.com/veler/DevToys/blob/main/CONTRIBUTING.md)
- [x] Verified that the change work in Release build configuration
- [x] Checked all unit tests pass

### CODE REVIEW

**Comment by veler** [comment]
Hello,
Thank you very much for this pull request. The cron tool is something that is quite asked by people and I'm sure this will please a lot of customers!

I like the UI overall. A few feedback about it:
The `Next scheduled dates` doesn't scale vertically when the window is being resized. It would be nice to support it like `Regex Tester` tool (you can take example on its implementation ;-) ) 
https://github.com/veler/DevToys/blob/main/src/dev/impl/DevToys/Views/Tools/Text/RegEx/RegExToolPage.xaml

image

Also, the UI feels crowded when we turn on the Compact Overlay (Picture in Picture) mode:
image

Perhaps we could hide the options in this scenario? It's something we do in many tools like JSON Formatter. You can use VisualState for this. Here is an example: https://github.com/veler/DevToys/blob/ca0b2f838ce5e5743d904430be0ac323a80ace6f/src/dev/impl/DevToys/Views/Tools/Formatters/JsonFormatter/JsonFormatterToolPage.xaml#L21

Thank you again :) This is a great PR !

**Comment by DotNetInAction** [comment]
Thank for the review and comments!

I have update my view:

1. Bottom TextBox now fills up the empty space on any size window
2. I have made settings collapsable in compact mode

Let me know what you think!

**Comment by DotNetInAction** [comment]
Sorry @veler, i thought those two UI changes are the only feedback, unless im being blind :D

**Review by veler — CHANGES_REQUESTED** [review]
No worries. As I mentioned, I left a few other comments in the code itself. :)

File: src/dev/impl/DevToys/Assets/Icons/Cron.svg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48">
```
Comment (veler) [inline src/dev/impl/DevToys/Assets/Icons/Cron.svg:1]: Where is this SVG coming from? Is there a license we should be aware of?

**Review by veler — APPROVED** [review]
Alright, I addressed all the issues. Thanks again for this contribution. It will be available in the next update. 😁

**Pull request merged**